h2o-core/src/main/java/water/api/ModelMetricsHandler.java

```diff
@@ -352,50 +352,52 @@ public JobV3 predictAsync(int version, final ModelMetricsListSchemaV3 s) {
     if (null == s.frame) throw new H2OIllegalArgumentException("frame", "predict", s.frame);
     if (null == DKV.get(s.frame.name)) throw new H2OKeyNotFoundArgumentException("frame", "predict", s.frame.name);
 
-    if (s.deviances || null != s.deviances_frame) throw new H2OIllegalArgumentException("deviances", "not supported for async", s.deviances_frame);
+    if (s.deviances || null != s.deviances_frame) 
+      throw new H2OIllegalArgumentException("deviances", "not supported for async", s.deviances_frame);
 
     final ModelMetricsList parms = s.createAndFillImpl();
-
-    //predict2 does not return modelmetrics, so cannot handle deeplearning: reconstruction_error (anomaly) or GLRM: reconstruct and archetypes
-    //predict2 can handle deeplearning: deepfeatures and predict
-
-    if (s.deep_features_hidden_layer > 0 || s.deep_features_hidden_layer_name != null) {
+    
+    long workAmount = -1;
```

Comment:
why not set the default to parms._frame.anyVec().nChunks() and get rid of the tertiary op at the end?

Reply:
:) no idea, thanks - fixed